### PR TITLE
add support for Quark_Plus

### PR DESCRIPTION
**Pull request by npspears**
## Description

Add new keyboard, Quark_Plus

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/checkerboards/quark_plus/readme.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+The Quark Plus is a premium Quark for those that want more features of the original. It features support for 4 rotary encoders (it is recommended to use 1 per left and right side), LED backlighting, RGB underglow, JST connector, and compatibility for earlier revisions of your favorite ortholinear keyboard.
+
+* Keyboard Maintainer: [Nasp](https://github.com/npspears)
+* Hardware Supported: Quark_Plus
+* Hardware Availability: https://www.checkerboards.xyz
+
+Make example for this keyboard (after setting up your build environment):
+
+    make quark_plus:default
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/readme.md:13]:
```suggestion
    make checkerboards/quark_plus:default
```

File: keyboards/checkerboards/quark_plus/keymaps/2u/keymap.c [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  * | TAB       |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | DEL   |
+  * |------------+------+------+------+------+-------------+------+------+------+------+------|
+  * | CTRL&ESC  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "    |
+  * |------------+------+------+------+------+------|------+------+------+------+------+------|
+  * | SHIFT     |   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  | ENTER |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+------|
+  * |    PRINT   | Ctrl | ALT  | GUI  |LOWER |   SPACE    |RAISE | LEFT | DOWN |  UP  |RIGHT  |
+  * `-----------------------------------------------------------------------------------------'
+  */
+  [0] = LAYOUT_2u( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/2u/keymap.c:38]:
```suggestion
  [0] = LAYOUT_2u(
```
etc.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by npspears** [comment]
thanks for the quick review, fauxpark

**Review by ridingqwerty — APPROVED** [review]

File: keyboards/checkerboards/quark_plus/readme.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Quark_Plus
+
+![Quark_Plus](https://i.imgur.com/4TI6BA0.png)
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/readme.md:3]:
```suggestion
![Quark_Plus](https://i.imgur.com/4TI6BA0h.png)
```

File: keyboards/checkerboards/quark_plus/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  * |       |      |      |  UP  |       |      |     |   _   |     |   [   |  ]    |      |
+  * |------------+------+------+------+------+-------------+------+------+------+------+---|
+  * |       |     | LEFT | DOWN  | RIGHT |     |      |   -  |      |   [   |   ]   |      |
+  * |------------+------+------+------+------+-----+-----+------+------+------+------+-----|
+  * |   F1  | F2 |  F3  |   F4  | F5    | F6  | F7   | F8   | F9    | F10  | F11   | F12   |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+---|
+  * |RESET |    |      |       |       |     |      |      |       |      |       |        |
+  * `---------------------------------------------------------------------------------------'
+     */
+  [2] = LAYOUT_grid( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/via/keymap.c:76]:
```suggestion
  [2] = LAYOUT_grid(
```

File: keyboards/checkerboards/quark_plus/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  * |   ~    | !    |  @   |  #  |  $   |  %   |  | ^  |  &   |  *  |  (    |  )  | BCKSPC  |
+  * |------------+------+------+------+------+-------------+------+------+------+------+----|
+  * |   \    |    1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  |   /   |
+  * |------------+------+------+------+------+------|------+------+------+------+------+----|
+  * |       |       |      |      |      |      |      |   +   |   =  |      |      |       |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+----|
+  * |     |    | ESC  |CTRL-ALT-DEL|TASK|    |     |     |     |    |  '|'  |   `   |       |
+  * `---------------------------------------------------------------------------------------'
+  */
+  [1] = LAYOUT_grid( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/via/keymap.c:57]:
```suggestion
  [1] = LAYOUT_grid(
```

File: keyboards/checkerboards/quark_plus/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  * | TAB       |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | DEL   |
+  * |------------+------+------+------+------+-------------+------+------+------+------+------|
+  * | CTRL&ESC  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "    |
+  * |------------+------+------+------+------+------|------+------+------+------+------+------|
+  * | SHIFT     |   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  | ENTER |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+------|
+  * |    PRINT   | Ctrl | ALT  | GUI  |LOWER |SPACE |SPACE|RAISE | LEFT | DOWN |  UP  |RIGHT  |
+  * `-----------------------------------------------------------------------------------------'
+  */
+  [0] = LAYOUT_grid( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/via/keymap.c:38]:
```suggestion
  [0] = LAYOUT_grid(
```

File: keyboards/checkerboards/quark_plus/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  * |       |      |      |  UP  |       |      |     |   _   |     |   [   |  ]    |      |
+  * |------------+------+------+------+------+-------------+------+------+------+------+---|
+  * |       |     | LEFT | DOWN  | RIGHT |     |      |   -  |      |   [   |   ]   |      |
+  * |------------+------+------+------+------+-----+-----+------+------+------+------+-----|
+  * |   F1  | F2 |  F3  |   F4  | F5    | F6  | F7   | F8   | F9    | F10  | F11   | F12   |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+---|
+  * |RESET |    |      |       |       |     |      |      |       |      |       |        |
+  * `---------------------------------------------------------------------------------------'
+     */
+  [2] = LAYOUT_grid( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/default/keymap.c:76]:
```suggestion
  [2] = LAYOUT_grid(
```

File: keyboards/checkerboards/quark_plus/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  * |   ~    | !    |  @   |  #  |  $   |  %   |  | ^  |  &   |  *  |  (    |  )  | BCKSPC  |
+  * |------------+------+------+------+------+-------------+------+------+------+------+----|
+  * |   \    |    1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  |   /   |
+  * |------------+------+------+------+------+------|------+------+------+------+------+----|
+  * |       |       |      |      |      |      |      |   +   |   =  |      |      |       |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+----|
+  * |     |    | ESC  |CTRL-ALT-DEL|TASK|    |     |     |     |    |  '|'  |   `   |       |
+  * `---------------------------------------------------------------------------------------'
+  */
+  [1] = LAYOUT_grid( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/default/keymap.c:57]:
```suggestion
  [1] = LAYOUT_grid(
```

File: keyboards/checkerboards/quark_plus/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  * | TAB       |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | DEL   |
+  * |------------+------+------+------+------+-------------+------+------+------+------+------|
+  * | CTRL&ESC  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "    |
+  * |------------+------+------+------+------+------|------+------+------+------+------+------|
+  * | SHIFT     |   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  | ENTER |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+------|
+  * |    PRINT   | Ctrl | ALT  | GUI  |LOWER |SPACE |SPACE|RAISE | LEFT | DOWN |  UP  |RIGHT  |
+  * `-----------------------------------------------------------------------------------------'
+  */
+  [0] = LAYOUT_grid( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/default/keymap.c:38]:
```suggestion
  [0] = LAYOUT_grid(
```

File: keyboards/checkerboards/quark_plus/keymaps/2x3u/keymap.c [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+* |       |      |      |  UP  |       |      |     |   _   |     |   [   |  ]    |      |
+* |------------+------+------+------+------+-------------+------+------+------+------+---|
+* |       |     | LEFT | DOWN  | RIGHT |     |      |   -  |      |   [   |   ]   |      |
+* |------------+------+------+------+------+-----+-----+------+------+------+------+-----|
+* |   F1  | F2 |  F3  |   F4  | F5    | F6  | F7   | F8   | F9    | F10  | F11   | F12   |
+* |-------+-------+-------+-------+-------+-------+------+------+------+------+------+---|
+* |     RESET    |           |          |         |         |         |        |         |
+* `---------------------------------------------------------------------------------------'
+   */
+[2] = LAYOUT_2x3u( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/2x3u/keymap.c:76]:
```suggestion
[2] = LAYOUT_2x3u(
```

File: keyboards/checkerboards/quark_plus/keymaps/2x3u/keymap.c [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+* |   ~    | !    |  @   |  #  |  $   |  %   |  | ^  |  &   |  *  |  (    |  )  | BCKSPC  |
+* |------------+------+------+------+------+-------------+------+------+------+------+----|
+* |   \    |    1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  |   /   |
+* |------------+------+------+------+------+------|------+------+------+------+------+----|
+* |     |      |      |      |      |      |      |   +   |   =   |      |        |       |
+* |-------+-------+-------+-------+-------+-------+------+------+------+------+------+----|
+* |   ESC  |  CTRL-ALT-DEL   |   TASK   |         |        |    '|'    |    `    |        |
+* `---------------------------------------------------------------------------------------'
+*/
+[1] = LAYOUT_2x3u( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/2x3u/keymap.c:57]:
```suggestion
[1] = LAYOUT_2x3u(
```

File: keyboards/checkerboards/quark_plus/keymaps/2x3u/keymap.c [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+* | Tab        |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Del  |
+* |------------+------+------+------+------+-------------+------+------+------+------+------|
+* | CTL & ESC  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "   |
+* |------------+------+------+------+------+------|------+------+------+------+------+------|
+* | Shift      |   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |Enter |
+* |-------+-------+-------+-------+-------+-------+------+------+------+------+------+------|
+* | PRINT   |  OS   |  Alt  |  Layer  |    Space & Layer     |   [     |   ]    |    CAPS   |
+* `-----------------------------------------------------------------------------------------'
+*/
+[0] = LAYOUT_2x3u( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/2x3u/keymap.c:38]:
```suggestion
[0] = LAYOUT_2x3u(
```

File: keyboards/checkerboards/quark_plus/keymaps/2x225u/keymap.c [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  * |       |      |      |  UP  |       |      |     |   _   |     |   [   |  ]    |      |
+  * |------------+------+------+------+------+-------------+------+------+------+------+---|
+  * |       |     | LEFT | DOWN  | RIGHT |     |      |   -  |      |   [   |   ]   |      |
+  * |------------+------+------+------+------+-----+-----+------+------+------+------+-----|
+  * |   F1  | F2 |  F3  |   F4  | F5    | F6  | F7   | F8   | F9    | F10  | F11   | F12   |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+---|
+  * |     RESET    |           |          |         |         |         |        |         |
+  * `---------------------------------------------------------------------------------------'
+     */
+  [2] = LAYOUT_2x225u( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/2x225u/keymap.c:76]:
```suggestion
  [2] = LAYOUT_2x225u(
```

File: keyboards/checkerboards/quark_plus/keymaps/2x225u/keymap.c [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+  * |   ~    | !    |  @   |  #  |  $   |  %   |  | ^  |  &   |  *  |  (    |  )  | BCKSPC  |
+  * |------------+------+------+------+------+-------------+------+------+------+------+----|
+  * |   \    |    1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  |   /   |
+  * |------------+------+------+------+------+------|------+------+------+------+------+----|
+  * |     |      |      |      |      |      |      |   +   |   =   |      |        |       |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+----|
+  * |   ESC  |  CTRL-ALT-DEL   |   TASK   |         |        |    '|'    |    `    |        |
+  * `---------------------------------------------------------------------------------------'
+  */
+  [1] = LAYOUT_2x225u( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/2x225u/keymap.c:57]:
```suggestion
  [1] = LAYOUT_2x225u(
```

File: keyboards/checkerboards/quark_plus/keymaps/2x225u/keymap.c [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+  * | Tab        |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Del  |
+  * |------------+------+------+------+------+-------------+------+------+------+------+------|
+  * | CTL & ESC  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "   |
+  * |------------+------+------+------+------+------|------+------+------+------+------+------|
+  * | Shift      |   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |Enter |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+------|
+  * | PRINT   |  OS   |  Alt  |  Layer  |    Space & Layer     |   [     |   ]    |    CAPS   |
+  * `-----------------------------------------------------------------------------------------'
+  */
+  [0] = LAYOUT_2x225u( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/2x225u/keymap.c:38]:
```suggestion
  [0] = LAYOUT_2x225u(
```

File: keyboards/checkerboards/quark_plus/keymaps/2u/keymap.c [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+  * |       |      |      |  UP  |       |      |     |   _   |     |   [   |  ]    |      |
+  * |------------+------+------+------+------+-------------+------+------+------+------+---|
+  * |       |     | LEFT | DOWN  | RIGHT |     |      |   -  |      |   [   |   ]   |      |
+  * |------------+------+------+------+------+-----+-----+------+------+------+------+-----|
+  * |   F1  | F2 |  F3  |   F4  | F5    | F6  | F7   | F8   | F9    | F10  | F11   | F12   |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+---|
+  * |RESET |    |      |       |       |            |      |       |      |       |        |
+  * `---------------------------------------------------------------------------------------'
+     */
+  [2] = LAYOUT_2u( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/2u/keymap.c:76]:
```suggestion
  [2] = LAYOUT_2u(
```

File: keyboards/checkerboards/quark_plus/keymaps/2u/keymap.c [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+  * |   ~    | !    |  @   |  #  |  $   |  %   |  | ^  |  &   |  *  |  (    |  )  | BCKSPC  |
+  * |------------+------+------+------+------+-------------+------+------+------+------+----|
+  * |   \    |    1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  |   /   |
+  * |------------+------+------+------+------+------|------+------+------+------+------+----|
+  * |       |       |      |      |      |      |      |   +   |   =  |      |      |       |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+----|
+  * |     |    | ESC  |CTRL-ALT-DEL|TASK|        |     |     |    |  '|'  |   `   |       |
+  * `---------------------------------------------------------------------------------------'
+  */
+  [1] = LAYOUT_2u( \
```
Comment (fauxpark) [inline keyboards/checkerboards/quark_plus/keymaps/2u/keymap.c:57]:
```suggestion
  [1] = LAYOUT_2u(
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**